src/main/webapp/user_script.js

```diff
@@ -114,3 +111,50 @@ function createFavoriteItemCard(entertainmentItem) {
 
   return card;
 }
+
+/**
+ * Loads the recommended users and displays them in a scrollable div.
+ *
+ * @param { Array } recommendedUsers - list with the recommended users in
+ *     descending order.
+ */
+function loadRecommendedUsers(recommendedUsers) {
+  const usersContainer = $('#recommendedUsersContainer');
+  const usersRow = $('<div class="row"></div>');
+  usersContainer.append(usersRow);
+
+  recommendedUsers.forEach((email) => {
+    fetch('/profile-data?email=' + email)
+        .then((response) => response.json())
+        .then((profile) => {
+          const userCol = $('<div class="col col-md-3"></div>');
+          userCol.append(createProfileCard(profile));
+
+          usersRow.append(userCol);
+        })
+        .catch((error) => {
+          console.log('Failed to fetch data for profile: ' + email);
+        });
+  });
+}
+
+/**
+ * Creates a card div that displays the user avatar and the username.
+ *
+ * @param { JSON } profile - the JSON object holding data about a user profile
+ * @returns { jQuery } card element representing the user
+ */
+function createProfileCard(profile) {
+  const card = $('<div class="card bg-light"></div>');
+  card.append(
+      $('<img class="card-img-top" src="https://icotar.com/avatar/' +
+        profile.username.charAt(0) + '">'));
```

Comment:
Lets move the avatar url logic  `"https://icotar.com/avatar/" + profile.username.charAt(0)`  into a helper function and call that function in loadProfile() line 31

This guarantees that all parts of the codebase use the same logic to create the URLs.

Reply:
OK I separated the avatar url logic into a separate function in last commit.